runtime-tests/tests/project_registration.rs

```diff
@@ -21,127 +21,261 @@
+        submit_ok(&client, &author, message.clone()).await;
+
+        // Duplicate submission with a different metadata.
+        let registration_2 = submit_ok(
+            &client,
+            &author,
+            message::RegisterProject {
+                metadata: Bytes128::random(),
+                ..message.clone()
+            },
+        )
+        .await;
+
+        assert_eq!(
+            registration_2.result,
+            Err(RegistryError::DuplicateProjectId.into())
+        );
+
+        let project = client
+            .get_project(message.project_name, message.project_domain)
+            .await
+            .unwrap()
+            .unwrap();
+        // Assert that the project data was not altered during the
+        // attempt to re-register the already existing project.
+        assert_eq!(message.metadata, project.metadata);
+
+        let projects_list = match &domain {
+            ProjectDomain::Org(org_id) => {
+                let org = client.get_org(org_id.clone()).await.unwrap().unwrap();
+                org.projects
+            }
+            ProjectDomain::User(user_id) => {
+                let user = client.get_user(user_id.clone()).await.unwrap().unwrap();
+                user.projects
+            }
+        };
+
+        // Assert that the number of projects in the involved domain didn't change.
+        assert_eq!(projects_list.len(), 1);
+        assert!(
+            projects_list.contains(&project.name),
+            format!(
+                "Registered project not found in the project list of {}",
+                domain
+            )
+        );
+    }
 }
 
+// Verify that two different orgs can have a project identified by the same name.
 #[async_std::test]
-async fn register_project_under_inexistent_org() {
+async fn re_register_project_org_domain() {
     let (client, _) = Client::new_emulator();
     let (author, _) = key_pair_with_associated_user(&client).await;
+    let domain_org_1 = ProjectDomain::Org(register_random_org(&client, &author).await.id);
 
-    let project_hash = H256::random();
     let checkpoint_id = submit_ok(
         &client,
         &author,
         message::CreateCheckpoint {
-            project_hash,
+            project_hash: H256::random(),
             previous_checkpoint_id: None,
         },
     )
     .await
     .result
     .unwrap();
 
-    let inexistent_org_id = random_id();
-    let message =
-        random_register_project_message(&ProjectDomain::Org(inexistent_org_id), checkpoint_id);
-    let tx_included = submit_ok(&client, &author, message.clone()).await;
+    let message = random_register_project_message(&domain_org_1, checkpoint_id);
+    submit_ok(&client, &author, message.clone()).await;
 
-    assert_eq!(tx_included.result, Err(RegistryError::InexistentOrg.into()));
+    // Duplicate submission under another domain.
+    let domain_org_2 = ProjectDomain::Org(register_random_org(&client, &author).await.id);
```

Comment:
I find it easier to follow if you split async side effects
```suggestion
    let org = register_random_org(&client, &author).await;
    let domain_org_2 = ProjectDomain::Org(org.id);
```